paymentsheet/src/main/java/com/stripe/android/paymentsheet/IntentConfirmationHandler.kt

```diff
@@ -101,8 +95,13 @@ internal class IntentConfirmationHandler(
     val hasReloadedFromProcessDeath = hasReloadedWhileAwaitingPreConfirm || hasReloadedWhileAwaitingConfirm
 
     private val _state = MutableStateFlow(
-        if (hasReloadedFromProcessDeath) {
-            State.Confirming(isPaymentSheetVisible)
+        if (hasReloadedWhileAwaitingPreConfirm) {
+            State.Preconfirming(
+                confirmationOption = currentArguments?.confirmationOption,
+                inPreconfirmFlow = true,
+            )
+        } else if (hasReloadedWhileAwaitingConfirm) {
+            State.Confirming
```

Comment:
could we simplify this by just having a SavedStateHandle that saves the `State`? Instead of having multiple and then needing to do the computation here

Reply:
Considered doing this initially but it could be problematic to save a `Complete` state. If we come back from process death and there's still a `State.Complete` value, we might re-emit a payment status that we have already emitted.

Reply:
gotcha, makes sense